src/java.base/share/classes/java/util/zip/ZipUtils.java

```diff
@@ -170,23 +171,31 @@ static LocalDateTime javaEpochToLocalDateTime(long time) {
      * The bytes are assumed to be in Intel (little-endian) byte order.
      */
     public static final int get16(byte[] b, int off) {
```

Comment:
Can JIT automatically perform MergeStore here? If JIT can automatically perform MergeStore without using Unsafe, many scenarios will benefit.

Reply:
Unfortunately we only have merge store for writing to arrays; no merge load when we are reading from arrays.

Reply:
Yes. As I wrote in the PR description: "An appealing alternative would be to implement a merge load analogue to the merge store optimizations in C2." So it's the opposite of a merge store. I recall @eme64 mentioning somewhere that such an optimization would be possible, if we could show it to be profitable. I think it's reasonable to do this enhancement now, then revert the `Unsafe` stuff if/when there's a merge load optimization that beats it.